packages/oauth/src/utils/urls.ts

```diff
@@ -50,8 +50,35 @@ export function logoutRedirect(redirectUri: URL): void {
   })
 }
 
+function validateRedirectURI(redirectUri: URL) {
+  const currentUrl = new URL(window.location.href)
+  const { hostname, port, protocol } = redirectUri
+  return (
+    hostname === currentUrl.hostname &&
+    port === currentUrl.port &&
+    protocol === currentUrl.protocol &&
+    ['http:', 'https:'].includes(protocol)
+  )
+}
+
+function sanitizeUri(url: URL) {
+  const searchParams = url.searchParams
+
+  if (searchParams.toString() !== '') {
+    searchParams.forEach((value, key) => {
+      searchParams.set(key, encodeURIComponent(value))
+    })
+  }
+  return url.href
+}
+
 export function redirect(target: URL) {
   log.debug('Redirecting to URI:', target)
   // Redirect to the target URI
+  if (validateRedirectURI(target)) {
+    window.location.href = sanitizeUri(target)
+  } else {
+    log.error('Invalid redirectUrl parameter')
```

Comment:
I'd say if the redirect is invalid then perhaps throwing an Error would be beneficial since something really would have gone wrong. It could get lost in the error log. Indeed, we'd also want to stop the process from going any further, especially allowing the invalid url to be run unsanitized?

Reply:
I will throw the error. I believe we are essentially doing that. Uri will be unsatinized only if it doesn't have any parameters which is fine.  To proceed with redirect, the URL has to be valid at first, so in this case protocol host and port is the same as host.  (Have to be tested though)

Reply:
Updated